scanners/zap/zap_local.py

```diff
@@ -0,0 +1,172 @@
+import logging
+import os
+import pprint
+import shutil
+import subprocess
+
+from .zap import PathIds
+from .zap import Zap
+from scanners import State
+
+CLASSNAME = "ZapLocal"
+
+
+pp = pprint.PrettyPrinter(indent=4)
+
+# Helper: absolute path to this directory (which is not the current directory)
+# Useful for finding files in this directory
+MODULE_DIR = os.path.dirname(__file__)
+
+
+class ZapLocal(Zap):
+    ###############################################################
+    # PRIVATE CONSTANTS                                           #
+    # Accessed by ZapLocalhost only                               #
+    ###############################################################
+
+    ###############################################################
+    # PROTECTED CONSTANTS                                         #
+    # Accessed by parent Zap object                               #
+    ###############################################################
+
+    def __init__(self, config):
+        """Initialize all vars based on the config.
+        The code of the function only deals with the "podman" layer, the "ZAP" layer is handled by super()
+        """
+
+        logging.debug("Initializing podman-based ZAP scanner")
+        super().__init__(config)
+
+        # prepare the host <-> container mapping
+        # Because there's no container layer, there's no need to translate anything
+        temp_dir = self._create_work_dir()
+        policies_dir = f"{os.environ['HOME']}/.ZAP/policies"
+
+        self.path_map.add(PathIds.WORK, temp_dir, temp_dir)
+        self.path_map.add(
+            PathIds.SCRIPTS, f"{MODULE_DIR}/scripts", f"{MODULE_DIR}/scripts"
+        )
+        self.path_map.add(PathIds.POLICIES, policies_dir, policies_dir)
+
+    ###############################################################
+    # PUBLIC METHODS                                              #
+    # Accessed by RapiDAST                                        #
+    # + MUST be implemented                                       #
+    # + SHOUT call super().<method>                               #
+    # + list: setup(), run(), postprocess(), cleanup()            #
+    ###############################################################
+
+    def setup(self, executable=None):
+        """Prepares everything:
+        - the command line to run
+        - environment variables
+        - files & directory
+
+        The code of the function only deals with the "localhost" layer, the "ZAP" layer is handled by super()
+        """
+
+        if self.state != State.UNCONFIGURED:
+            raise RuntimeError(
+                f"Podman setup encounter an unexpected state: {self.state}"
+            )
+
+        super().setup(executable="zap.sh")
```

Comment:
I would propose the following, similar to what we do for `authentication` : 

```
zap:
  container:
    type: (podman|local)
    parameters:
      # for podman
      image:
      # for local
      exec:
```
`type` could be set in `general.container.type`, so that it will be merge and applied to all scanners, while `exec` or `image` can be set in each scanner

Reply:
However, this is a schema configuration change, so we will need to update the config version for that